Ensure the XProcLock is held when token object files are saved, read, or deleted.

ifranzki opened the pull request:
This fixes a problem with message `usr/lib/common/loadsave.c Cannot read size.` on syslog when running multi threaded.

Note: This patch is on top of the "Check XProcLock return values" patch.
Since that other patch is not yet merged, it is also included in this PR. 
I may need to rebase my patch once the other one is merged.

p-steuer commented:
Looks good to me. Maybe it would be a good idea to add a comment to functions like load_public_token_objects which require the caller to hold the lock.

ifranzki commented:
Added a note to the functions like load_public_token_objects in loadsave.c that they require the caller to hold the lock.
Rebased to the current master branch.
